co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureRegistrationReceiver.java

```diff
@@ -70,13 +70,14 @@ public void init() {
     @Override
     public void run() {
         byte[] buf = new byte[UDP_MTU];
-        running = true;
-        while (running) {
+        running.set(true);
+        while (running.get()) {
             DatagramPacket msg = new DatagramPacket(buf, buf.length);
             try {
                 listenSocket.receive(msg);
             } catch (IOException e) {
-                throw new RuntimeException(e);
```

Comment:
Throwing/Rethrowing this exception has the following issues. On each call to close, a SocketException will be triggered here since the `close` method is called outside the `run` while loop and there is currently no timeout configured and no sleep/periodic polling setup on the datagramsocket. This means that when stop is called and the socket is closed, the InfrastructureRegistrationReceiver thread is still blocked on the receive() call and then throws an exception because we have now closed the socket. The way I avoid this is on exception I continue and re-evaluate the running condition of the socket.